msbrainpy/map.py

```diff
@@ -179,3 +180,162 @@ def affineTransform(parameters, COR, points):
+    """
+    find_gene_series_masks(directory, out_directory, image_name_pattern=image_name_pattern)
+    volume_assembler = get_volume_assembler(out_directory, scale_factor=scale_factor)
+    return volume_assembler
+
+
+def get_volume_assembler(tissue_mask_directory, scale_factor=8, tissue_mask_pattern=r'\d*_tissue_mask.tif'):
+    dictionary = get_volume_assembler_dict(tissue_mask_directory, scale_factor=scale_factor,
+                                           tissue_mask_pattern=tissue_mask_pattern)
+    volume_assembler = VolumeAssembler(dictionary)
+    return volume_assembler
+
+
+def get_volume_assembler_dict(tissue_mask_directory, scale_factor=8, tissue_mask_pattern=r'\d*_tissue_mask.tif'):
+    # initialise the dictionary which will be filled with information about volume assembly (for VolumeAssembler object)
+    transformation_dictionary = {'coordinates_list' : []}
+    # find the files containing tissue masks
+    tissue_mask_files = []
+    tissue_mask_pattern = re.compile(tissue_mask_pattern)
+    for file in sorted(os.listdir(tissue_mask_directory)):
+        match = tissue_mask_pattern.match(file)
+        if match is not None:
+            tissue_mask_files.append(match[0])
+    # find the important parameters
+    for i in range(len(tissue_mask_files)):
+        file = tissue_mask_files[i]
+        file_path = os.path.join(tissue_mask_directory, file)
+        index = str(re.findall(r'\d*', file)[0])
+        mask_image = io.imread(file_path)
+        mask_image = mask_image.astype(int)
+        props = measure.regionprops(mask_image)[0]
+        bounding_box = (np.round(props['bbox'][0]*scale_factor).astype(int),
+                        np.round(props['bbox'][1]*scale_factor).astype(int),
+                        np.round(props['bbox'][2]*scale_factor).astype(int),
+                        np.round(props['bbox'][3]*scale_factor).astype(int))
+        centroid = (np.round(props['local_centroid'][0]*scale_factor).astype(int),
+                    np.round(props['local_centroid'][1]*scale_factor).astype(int) )
+        transformation_dictionary['coordinates_list'].append({})
+        transformation_dictionary['coordinates_list'][i]['index'] = index
+        transformation_dictionary['coordinates_list'][i]['bounding_box'] = bounding_box
+        transformation_dictionary['coordinates_list'][i]['local_centroid'] = centroid
+    return transformation_dictionary
+
+
+def get_volume_from_series(volume_assembler, series_pattern, directory, scale_factor=1):
+    files = []
+    pattern = re.compile(series_pattern)
+    for file in os.listdir(directory):
+        match = pattern.match(file)
+        if match is not None:
+            files.append(match[0])
+    volume = volume_assembler.generate_volume(directory, files)
+    return volume
+
+# -------------------------------------------- VolumeAssembler Class ---------------------------------------------------
+class VolumeAssembler:
```

Comment:
Do you have a clear idea in your head on when to use a class vs a function? Opinions differ on where the balance should lie, but I try to leave classes as a last resort. I'm not entirely sure this one is needed, since you have just a little bit of data storage (a dictionary, mostly), and then immediately use that to write it out, and are done. Generally, I keep classes for cases where you have to store several related bits of data together over a period of time.